src/content/blog/astro-530.mdx

```diff
@@ -0,0 +1,93 @@
+
+- [**Faster page rendering**](#faster-page-rendering)
+- [**Automatic session storage setup**](#automatic-session-storage-setup)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Faster page rendering
+
+Astro makes it easy to use async functions inside components, but previously this meant that every component and page was rendered asynchronously, even if it wasn't needed. When this was happening for every component on a page, the overhead from these promises would really add up. Now, thanks to an awesome community contribution from [MatthewLymer](https://github.com/MatthewLymer), Astro will only render components asynchronously if they actually contain an async function call. In benchmarks, this has resulted in a significant speedup for rendering synchronous components. This benefits both on-demand and static builds, though the biggest improvements are seen in SSR response times. This even helps pages that use async functions, because most components on the page will still be synchronous.
+
+The exact amount of the improvement will depend on the individual sites. Complex pages with many components will see the biggest improvements, while pages that use slower external APIs at request time may be unchanged. In our testing, SSR responses are generally 1.5-2x faster unless the page is querying external APIs. Some examples:
+
+- For the 5300-page [Astro docs site](https://docs.astro.build), it **cut static build times by 10-15%.** 
+- In the Platformic SSR benchmark, which generates a page with over 3000 components, requests are **4x faster**.
+
+You don't need to do anything to benefit from this improvement: just sit back and enjoy the faster rendering times.
+
+## Automatic session storage setup
+
+Astro 5.1 introduced the experimental [session storage API](https://docs.astro.build/en/reference/experimental-flags/sessions/), which allows developers to easily manage user sessions on the server. The feature supports dozens of different backend storage drivers, powered by [unstorage](https://unstorage.unjs.io). Previously you would need to configure the session storage driver in your `astro.config.mjs`, but now Astro will automatically set up the session storage for you when using a supported adapter. 
+
+Currently, these supported adapters include `@astrojs/node`, which uses the filesystem for storage, and `@astrojs/netlify`, which uses Netlify Blobs. We hope to add support for more adapters in the near future, where the hosting platforms include a storage API. In all cases, you can still manually configure your own session storage driver if you prefer.
+
+To enable session storage, set the `experimental.session` flag in your `astro.config.mjs`:
```

Comment:
```suggestion
If you haven't experimented with the session storage API yet, you can enable this feature by setting the `experimental.session` flag  to `true` in your `astro.config.mjs`:
```
Making a clearer split between who is adding this fresh and who needs to update a config